helix-term/src/main.rs

```diff
@@ -89,14 +90,26 @@ FLAGS:
 
     if args.health {
         if let Some(lang) = args.health_arg {
-            match lang.as_str() {
+            let io_result = match lang.as_str() {
                 "all" => helix_term::health::languages_all(),
                 _ => helix_term::health::language(lang),
+            };
+
+            if let Err(err) = io_result {
+                if err.kind() != std::io::ErrorKind::BrokenPipe {
+                    return Err(err.into());
+                }
             }
         } else {
-            helix_term::health::general();
-            println!();
-            helix_term::health::languages_all();
+            let io_result = helix_term::health::general()
+                .and_then(|_| writeln!(std::io::stdout().lock()))
+                .and_then(|_| helix_term::health::languages_all());
+
+            if let Err(err) = io_result {
+                if err.kind() != std::io::ErrorKind::BrokenPipe {
+                    return Err(err.into());
+                }
+            }
         }
```

Comment:
I think this can be extracted to a function like `print_health(lang: Option<String>) -> io::Result<()>` where we can use `?` abundantly and then handle `BrokenPipe` once in the calling code.

Reply:
I have created print_health function in health.rs. Now BrokenPipe is only handled once in main.rs